scripts/sct_pipeline.py

```diff
@@ -327,27 +328,26 @@ def run_function(function, folder_dataset, list_subj, list_args=[], nb_cpu=None,
 
     # Computing Pool for parallel process, distribute2mpi.MpiPool in MPI environment, multiprocessing.Pool otherwise
     sct.log.debug("stating pool with {} thread(s)".format(nb_cpu))
-    pool = Pool(nb_cpu)
+    pool = PoolExecutor(nb_cpu)
     compute_time = None
     try:
         compute_time = time()
         sct.log.debug('paused but print')
-        async_results = pool.map_async(function_launcher, list_func_subj_args)
-        pool.close()
-        pool.join()  # waiting for all the jobs to be done
+        # async_results = pool.map(function_launcher, list_func_subj_args)
+        async_results = [function_launcher(i) for i in list_func_subj_args]
         compute_time = time() - compute_time
-        all_results = async_results.get()
         # concatenate all_results into single Panda structure
-        results_dataframe = pd.concat([result for result in all_results])
+        #  async_results is an iterator that locks on __next__ call
```

Comment:
Same here...

Reply:
It still in WIP, I have found a specific point where it crashed. Working on it, hope to have it ready today.